<h1>Palmer Penguin Data Analytics<h1>
<hr>
<h3>Analyst : Ryan T Romero<h3>


In [324]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

In [325]:
penguins = pd.read_csv(r"C:\Users\Admin\Downloads\penguins.csv")
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1>Descriptive Analytics<h1>

In [326]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [327]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h1>Cleaning Data<h1>

In [328]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1> Data Frame</h1>

In [329]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [330]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [331]:
adelie_df = penguins[penguins['species']=="Adelie"]
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [332]:
gentoo_df = penguins[penguins['species']=="Gentoo"]
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [333]:
chinstrap_df = penguins[penguins['species']=="Chinstrap"]
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [334]:
penguins[penguins["body_mass_g"] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

In [335]:
penguins[(penguins['species']=="Chinstrap") & (penguins["body_mass_g"] >= 4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [336]:
penguins[(penguins['species']=="Chinstrap") & (penguins["body_mass_g"] >= 4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1>Filling up Null</h1>

In [337]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1> How to Change Data Types of a Column

In [338]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            344 non-null    object 
 1   island             344 non-null    object 
 2   bill_length_mm     344 non-null    float64
 3   bill_depth_mm      344 non-null    float64
 4   flipper_length_mm  344 non-null    float64
 5   body_mass_g        344 non-null    float64
 6   sex                344 non-null    object 
 7   year               344 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 21.6+ KB


In [339]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


<h1> Data analysis (7 data) </h1>

<h3>1. How many female chinstrap is there?</h3>

In [340]:
chinstrap_df[chinstrap_df["sex"] == 'female']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
281  Chinstrap  Dream            45.2           17.8              198.0   
282  Chinstrap  Dream            46.1           18.2              178.0   
284  Chinstrap  Dream            46.0           18.9              195.0   
286  Chinstrap  Dream            46.6           17.8              193.0   
288  Chinstrap  Dream            47.0           17.3              185.0   
290  Chinstrap  Dream            45.9           17.1              190.0   
293  Chinstrap  Dream            58.0           17.8              181.0   
294  Chinstrap  Dream            46.4           18.6              190.0   
296  Chinstrap  Dream            42.4           17.3              181.0   
298  Chinstrap  Dream            43.2           16.6              187.0   
300  Chinstrap  Dream            46.7           17.9              195.0   
302  Chinstrap  Dream            50.5           18.4              200.0   
304  Chinstrap  Dream            46.4           17.8              191.0   
306  Chinstrap  Dream            40.9           16.6              187.0   
308  Chinstrap  Dream            42.5           16.7              187.0   
311  Chinstrap  Dream            47.5           16.8              199.0   
312  Chinstrap  Dream            47.6           18.3              195.0   
314  Chinstrap  Dream            46.9           16.6              192.0   
317  Chinstrap  Dream            46.2           17.5              187.0   
319  Chinstrap  Dream            45.5           17.0              196.0   
320  Chinstrap  Dream            50.9           17.9              196.0   
322  Chinstrap  Dream            50.1           17.9              190.0   
325  Chinstrap  Dream            49.8           17.3              198.0   
326  Chinstrap  Dream            48.1           16.4              199.0   
328  Chinstrap  Dream            45.7           17.3              193.0   
330  Chinstrap  Dream            42.5           17.3              187.0   
332  Chinstrap  Dream            45.2           16.6              191.0   
335  Chinstrap  Dream            45.6           19.4              194.0   
337  Chinstrap  Dream            46.8           16.5              189.0   
338  Chinstrap  Dream            45.7           17.0              195.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
279       3525.0  female  2007  
281       3950.0  female  2007  
282       3250.0  female  2007  
284       4150.0  female  2007  
286       3800.0  female  2007  
288       3700.0  female  2007  
290       3575.0  female  2007  
293       3700.0  female  2007  
294       3450.0  female  2007  
296       3600.0  female  2007  
298       2900.0  female  2007  
300       3300.0  female  2007  
302       3400.0  female  2008  
304       3700.0  female  2008  
306       3200.0  female  2008  
308       3350.0  female  2008  
311       3900.0  female  2008  
312       3850.0  female  2008  
314       2700.0  female  2008  
317       3650.0  female  2008  
319       3500.0  female  2008  
320       3675.0  female  2009  
322       3400.0  female  2009  
325       3675.0  female  2009  
326       3325.0  female  2009  
328       3600.0  female  2009  
330       3350.0  female  2009  
332       3250.0  female  2009  
335       3525.0  female  2009  
337       3650.0  female  2009  
338       3650.0  female  2009  
340       3400.0  female  2009  
343       3775.0  female  2009

In [341]:
chinstrap_df[chinstrap_df["sex"] == 'female'].info()

<class 'pandas.core.frame.DataFrame'>
Index: 34 entries, 276 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            34 non-null     object 
 1   island             34 non-null     object 
 2   bill_length_mm     34 non-null     float64
 3   bill_depth_mm      34 non-null     float64
 4   flipper_length_mm  34 non-null     float64
 5   body_mass_g        34 non-null     float64
 6   sex                34 non-null     object 
 7   year               34 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 2.4+ KB


In [342]:
chinstrap_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 68 entries, 276 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            68 non-null     object 
 1   island             68 non-null     object 
 2   bill_length_mm     68 non-null     float64
 3   bill_depth_mm      68 non-null     float64
 4   flipper_length_mm  68 non-null     float64
 5   body_mass_g        68 non-null     float64
 6   sex                68 non-null     object 
 7   year               68 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.8+ KB


<h1>Insight</h1>
<h3>Out of 68 chinstrap entries, there's only a total of 34 female chinstrap.</h3>

<h3>2. How many gentoo is above 5000g? </h3>

In [343]:
gentoo_df[gentoo_df["body_mass_g"] >= 5000]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
153  Gentoo  Biscoe            50.0           16.3              230.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
159  Gentoo  Biscoe            46.7           15.3              219.0   
161  Gentoo  Biscoe            46.8           15.4              215.0   
..      ...     ...             ...            ...                ...   
267  Gentoo  Biscoe            55.1           16.0              230.0   
269  Gentoo  Biscoe            48.8           16.2              222.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
153       5700.0    male  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
159       5200.0    male  2007  
161       5150.0    male  2007  
..           ...     ...   ...  
267       5850.0    male  2009  
269       6000.0    male  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[67 rows x 8 columns]

In [344]:
gentoo_df[gentoo_df["body_mass_g"] >= 5000].info()

<class 'pandas.core.frame.DataFrame'>
Index: 67 entries, 153 to 275
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            67 non-null     object 
 1   island             67 non-null     object 
 2   bill_length_mm     67 non-null     float64
 3   bill_depth_mm      67 non-null     float64
 4   flipper_length_mm  67 non-null     float64
 5   body_mass_g        67 non-null     float64
 6   sex                67 non-null     object 
 7   year               67 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 4.7+ KB


In [345]:
gentoo_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 124 entries, 152 to 275
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            124 non-null    object 
 1   island             124 non-null    object 
 2   bill_length_mm     123 non-null    float64
 3   bill_depth_mm      123 non-null    float64
 4   flipper_length_mm  123 non-null    float64
 5   body_mass_g        123 non-null    float64
 6   sex                119 non-null    object 
 7   year               124 non-null    int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 8.7+ KB


<h1>Insights</h1>
<h3> Out of 124 gentoo data entries, only 67 of them weigh above 5000g.</h3>

<H3>3. What species has the most male?</H3>

In [346]:
chinstrap_df['sex'].value_counts()

sex
female    34
male      34
Name: count, dtype: int64

In [347]:
gentoo_df['sex'].value_counts()

sex
male      61
female    58
Name: count, dtype: int64

In [348]:
adelie_df['sex'].value_counts()

sex
male      73
female    73
Name: count, dtype: int64

<h1>Insights</h1>
<h3>From the given data, Adelie showed that it has the most male out of the three species.</h3>

<h3>4. What species has the longest bill length in average?

In [349]:
gentoo_df['bill_length_mm'].mean()

np.float64(47.50487804878048)

In [350]:
adelie_df['bill_length_mm'].mean()

np.float64(38.79139072847682)

In [351]:
chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

<h1>Insights</h1>
<h3>From the data collected, the species that has the longest average bill length is chinstrap with 48.83 mm.

<H3>5.What is the average bill depth of an adelie?</H3>

In [352]:
adelie_df['bill_depth_mm'].mean()

np.float64(18.346357615894043)

<h1>Insights</h1>
<h3>The average bill depth of an Adelie is 18.346 mm.</h3>

<h3>6. How many males and females are there in all?</h3>

In [353]:
penguins['sex'].value_counts()

sex
male      179
female    165
Name: count, dtype: int64

<h1>Insights</h1>
<h3>There are 179 males and 165 females with a total of 344 data entries.</h3>

<h3>7. What is the average flipper lenght of a Adelie?</h3>

In [354]:
adelie_df['flipper_length_mm'].mean()

np.float64(189.95364238410596)

<h1>Insights</h1>
<h3>From the given data, the average flipper lenght of the Adelie in mm is 189.95</h3>

<h1> Data Aggregation </h1>

<h3>How many species are there per island</h3>

In [355]:
penguins.groupby(['species','island']).count()

bill_length_mm  bill_depth_mm  flipper_length_mm  \
species   island                                                        
Adelie    Biscoe                 44             44                 44   
          Dream                  56             56                 56   
          Torgersen              52             52                 52   
Chinstrap Biscoe                  0              0                  0   
          Dream                  68             68                 68   
          Torgersen               0              0                  0   
Gentoo    Biscoe                124            124                124   
          Dream                   0              0                  0   
          Torgersen               0              0                  0   

                     body_mass_g  sex  year  
species   island                             
Adelie    Biscoe              44   44    44  
          Dream               56   56    56  
          Torgersen           52   52    52  
Chinstrap Biscoe               0    0     0  
          Dream               68   68    68  
          Torgersen            0    0     0  
Gentoo    Biscoe             124  124   124  
          Dream                0    0     0  
          Torgersen            0    0     0

In [356]:
penguins.groupby(['species','island','sex']).size()

species    island     sex   
Adelie     Biscoe     female    22
                      male      22
           Dream      female    27
                      male      29
           Torgersen  female    24
                      male      28
Chinstrap  Biscoe     female     0
                      male       0
           Dream      female    34
                      male      34
           Torgersen  female     0
                      male       0
Gentoo     Biscoe     female    58
                      male      66
           Dream      female     0
                      male       0
           Torgersen  female     0
                      male       0
dtype: int64

<h3>Show the collected data of all the species in each year </h3>

In [357]:
penguins.groupby(['species','year']).size()

species    year
Adelie     2007    50
           2008    50
           2009    52
Chinstrap  2007    26
           2008    18
           2009    24
Gentoo     2007    34
           2008    46
           2009    44
dtype: int64

In [358]:
penguins.groupby(adelie_df['year']).size()

year
2007.0    50
2008.0    50
2009.0    52
dtype: int64

In [359]:
penguins.groupby(['year','sex','island'])['body_mass_g'].mean()

year  sex     island   
2007  female  Biscoe       4345.238095
              Dream        3446.590909
              Torgersen    3475.000000
      male    Biscoe       5102.173913
              Dream        3902.083333
              Torgersen    3991.812865
2008  female  Biscoe       4225.806452
              Dream        3444.117647
              Torgersen    3518.750000
      male    Biscoe       5006.060606
              Dream        4114.705882
              Torgersen    4193.750000
2009  female  Biscoe       4403.571429
              Dream        3447.727273
              Torgersen    3193.750000
      male    Biscoe       5114.898575
              Dream        3935.227273
              Torgersen    3784.375000
Name: body_mass_g, dtype: float64

<h1>8. What is the average body mass of Chinstrap penguins with sex in mind?<h1>

In [360]:
chinstrap_df.groupby(['species','sex'])['body_mass_g'].mean()

species    sex   
Chinstrap  female    3527.205882
           male      3938.970588
Name: body_mass_g, dtype: float64

<h1>Insights</h1>
<h3>The average body mass of a female chinstrap is 3527.21g while male chinstrap is 3938.97g </h3>

<h1>9. Which year has the highest number of gentoo?</h3>

In [361]:
gentoo_df.groupby(['species', 'year']).size()

species  year
Gentoo   2007    34
         2008    46
         2009    44
dtype: int64

<h1> Insights </h1>
<h3> The year 2009 has the most number data entries of gentoo</h3>

<h1>10. How many Chinstrap are recorded in each island?<h1>


In [362]:
penguins.groupby(['species', 'island']).size()['Chinstrap']


island
Biscoe        0
Dream        68
Torgersen     0
dtype: int64

<h1>Insights</h1>
<h3> Only the dream island has a recorded data of a chinstrap which is 68 while the rest is zero</h3>

<h1>11. What is the bosy mass of each species? <h1>

In [363]:
penguins.groupby('species')['body_mass_g'].mean()


species
Adelie       3703.958910
Chinstrap    3733.088235
Gentoo       5068.965761
Name: body_mass_g, dtype: float64

<h1>Insights</h1>
<h3> Adelie has 3703.96g average weight, on the other hand, Chinstrap has an average weight of 3733.08g while gentoo has 5068.97g </h3>

<h1>12. How many adelie data were recorded in the year 2008?<h1>


In [364]:
adelie_df.groupby(['species', 'year']).size()

species  year
Adelie   2007    50
         2008    50
         2009    52
dtype: int64

<h1>Insights</h1>
<h3>A total of 50 data entries were recorded in 2009 and the same goes for the year 2007</h3>